crates/cacache/src/content/write.rs

```diff
@@ -1,54 +1,106 @@
 use std::fs::DirBuilder;
-use std::io::prelude::*;
-use std::path::{Path, PathBuf};
-use std::pin::Pin;
-use std::sync::Mutex;
-
-use async_std::fs as afs;
-use async_std::future::Future;
-use async_std::task::{self, Context, JoinHandle, Poll};
-use futures::io::AsyncWrite;
-use futures::prelude::*;
+use std::io::Seek;
+use std::io::Write;
+use std::path::PathBuf;
+
 use memmap::MmapMut;
 use ssri::{Algorithm, Integrity, IntegrityOpts};
 use tempfile::NamedTempFile;
 
 use crate::content::path;
 use crate::errors::{Internal, Result};
 
-pub const MAX_MMAP_SIZE: usize = 1024 * 1024;
+pub const MAX_MMAP_WRITE_SIZE: usize = 1024 * 1024 * 10;
+pub const MIN_MMAP_WRITE_SIZE: usize = 1024 * 1024;
```

Comment:
These bounds need to be tested too. I suspect they're different between *nix, macos, and windows!